inflate.c

```diff
@@ -1070,7 +1070,7 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
     if (INFLATE_NEED_UPDATEWINDOW(strm) &&
             (state->wsize || (out != strm->avail_out && state->mode < BAD &&
                  (state->mode < CHECK || flush != Z_FINISH)))) {
-        if (updatewindow(strm, strm->next_out, out - strm->avail_out)) {
+        if (updatewindow(strm, strm->next_out, out - strm->avail_out, state->wrap & 4)) {
```

Comment:
A comment above this might be like `update sliding window with checksum only if using gzip or adler32`

Reply:
Isn't that all cases?  With the except of maybe raw mode?